### PR TITLE
feat: add `esm` bundle to `@stylexjs/stylex`

### LINKED ISSUE #205: [Feature Request] Provide ESM bundle
**The problem**
```shell
4:18:25 PM [vite] Error when evaluating SSR module /@fs/Users/himself65/Code/waku/node_modules/.pnpm/@stylexjs+stylex@0.3.0/node_modules/@stylexjs/stylex/lib/stylex.js:
|- ReferenceError: exports is not defined
    at eval (/Users/himself65/Code/waku/node_modules/.pnpm/@stylexjs+stylex@0.3.0/node_modules/@stylexjs/stylex/lib/stylex.js:5:23)
    at instantiateModule (file:///Users/himself65/Code/waku/node_modules/.pnpm/vite@5.0.9_@types+node@20.10.4/node_modules/vite/dist/node/chunks/dep-Pluk1iaB.js:50504:15)

4:18:25 PM [vite] Error when evaluating SSR module /src/components/Banner.tsx: failed to import "/@fs/Users/himself65/Code/waku/node_modules/.pnpm/@stylexjs+stylex@0.3.0/node_modules/@stylexjs/stylex/lib/stylex.js"
|- ReferenceError: exports is not defined
    at eval (/Users/himself65/Code/waku/node_modules/.pnpm/@stylexjs+stylex@0.3.0/node_modules/@stylexjs/stylex/lib/stylex.js:5:23)
    at instantiateModule (file:///Users/himself65/Code/waku/node_modules/.pnpm/vite@5.0.9_@types+node@20.10.4/node_modules/vite/dist/node/chunks/dep-Pluk1iaB.js:50504:15)

4:18:25 PM [vite] Error when evaluating SSR module /src/components/App.tsx: failed to import "/src/components/Banner.tsx"
|- ReferenceError: exports is not defined
    at eval (/Users/himself65/Code/waku/node_modules/.pnpm/@stylexjs+stylex@0.3.0/node_modules/@stylexjs/stylex/lib/stylex.js:5:23)
    at instantiateModule (file:///Users/himself65/Code/waku/node_modules/.pnpm/vite@5.0.9_@types+node@20.10.4/node_modules/vite/dist/node/chunks/dep-Pluk1iaB.js:50504:15)

ReferenceError: exports is not defined
    at eval (/Users/himself65/Code/waku/node_modules/.pnpm/@stylexjs+stylex@0.3.0/node_modules/@stylexjs/stylex/lib/stylex.js:5:23)
    at instantiateModule (file:///Users/himself65/Code/waku/node_modules/.pnpm/vite@5.0.9_@types+node@20.10.4/node_modules/vite/dist/node/chunks/dep-Pluk1iaB.js:50504:15)
ReferenceError: exports is not defined
    at eval (/Users/himself65/Code/waku/node_modules/.pnpm/@stylexjs+stylex@0.3.0/node_modules/@stylexjs/stylex/lib/stylex.js:5:23)
    at instantiateModule (file:///Users/himself65/Code/waku/node_modules/.pnpm/vite@5.0.9_@types+node@20.10.4/node_modules/vite/dist/node/chunks/dep-Pluk1iaB.js:50504:15) {
  digest: ''
}
Cannot render RSC ReferenceError: exports is not defined
    at eval (/Users/himself65/Code/waku/node_modules/.pnpm/@stylexjs+stylex@0.3.0/node_modules/@stylexjs/stylex/lib/stylex.js:5:23)
    at instantiateModule (file:///Users/himself65/Code/waku/node_modules/.pnpm/vite@5.0.9_@types+node@20.10.4/node_modules/vite/dist/node/chunks/dep-Pluk1iaB.js:50504:15) {
  digest: ''
}
```

**How to reproduce**

Steps to reproduce:
1.
2.
3.

**Expected behavior**

**Environment (include versions). Did this work in previous versions?**

### PR DESCRIPTION
**Pull request by nedjulius**
## What changed / motivation ?

This PR modifies the `build` script of `stylex` and adds `ECMAScript` module support for the package.

## Linked PR/Issues

Fixes #205 

## Pre-flight checklist

- [x] I have read the contributing guidelines
      [Contribution Guidelines](./CONTRIBUTING.md)
- [x] Performed a self-review of my code

### CODE REVIEW

**Comment by nmn** [comment]
This looks good so far. LMK, when you think it's ready for review.

**Comment by nedjulius** [comment]
@nmn yeah, I think it's ready now!

File: packages/stylex/package.json [hunk 1]
```diff
@@ -2,13 +2,28 @@
...
+  "types": "./lib/stylex.d.ts",
+  "exports": {
+    ".": {
+      "import": "./lib/es/stylex.js",
+      "require": "./lib/stylex.js",
+      "types": "./lib/stylex.d.ts"
+    },
+    "./lib/StyleXCSSTypes": "./lib/StyleXCSSTypes.js",
+    "./lib/StyleXSheet": "./lib/StyleXSheet.js",
+    "./lib/StyleXTypes": "./lib/StyleXTypes.js",
```
Comment (necolas) [inline packages/stylex/package.json:16]: @nmn can we avoid exporting these files?

Comment (nmn) [reply packages/stylex/package.json:16]: In some rare cases, `StyleXTypes` might be useful to export, but outside of that only `stylex.js` and `stylex-inject.js` are needed.

Comment (nmn) [reply packages/stylex/package.json:16]: Do you think we need the commonJS exports?

Comment (nedjulius) [reply packages/stylex/package.json]: removed the `StyleXCSSTypes` and `StyleXSheet` exports

Comment (nedjulius) [reply packages/stylex/package.json]: looks like `lib/StyleXSheet` export is used in `dev-runtime` package. I will add it back

**Comment by nmn** [comment]
Thank you @nedjulius! It looks good to merge, but I'm checking some things and will merge in a few days. I'll make sure to include this in the next point release.

**Comment by himself65** [comment]
I think this not enough, the upstream package you are using is still cjs. Which will still break the build

**Comment by nmn** [comment]
@himself65 Yes, I'm aware which is what I'm going to be fixing over the next few days!

**Comment by nedjulius** [comment]
@nmn are you looking into fixing the upstream package issue? I can try to look into it as well so we can unblock this PR

**Comment by nmn** [comment]
@nedjulius Yes, I have a PR up already. There's a couple of errors I hope to fix soon.

**Pull request merged**